packages/core/src/controllers/TransactionsController.ts

```diff
@@ -0,0 +1,93 @@
+import { DateUtil, type Transaction } from '@web3modal/common'
+import { proxy, subscribe as sub } from 'valtio/vanilla'
+import { BlockchainApiController } from './BlockchainApiController.js'
+import { OptionsController } from './OptionsController.js'
+import { AccountController } from './AccountController.js'
+
+// -- Types --------------------------------------------- //
+type TransactionByYearMap = Record<number, Transaction[]>
+
+export interface TransactionsControllerState {
+  transactions: Transaction[]
+  transactionsByYear: TransactionByYearMap
+  loading: boolean
+  empty: boolean
+  next: string | null
```

Comment:
```suggestion
  next?: string
```

We try to stay away from nulls and prefer `undefined`